packages/block-editor/src/components/block-list/index.native.js

```diff
@@ -93,6 +94,7 @@ export class BlockList extends Component {
 					extraScrollHeight={ innerToolbarHeight + 10 }
 					keyboardShouldPersistTaps="always"
 					scrollViewStyle={ { flex: isRootList ? 1 : 0 } }
+					containerStyle={ containerStyle }
```

Comment:
Pass `containerStyle` to `FlatList` to set proper style on `Columns` block and allow to render its column in `flex-direction: row`